telegraf crashes with panic (segmentation violation)

### System info:
crashes with 1.12.1, tried upgrading to 1.12.2 with same behaviour

Debian Stretch
Proxmox PVE kernel (adapted from ubuntu)
AMD Threadripper (first gen)
128 GB RAM

No recent hardware or software updates, except telegraf version

### Steps to reproduce:
Start telegraf
crashes in a couple of minutes
restarted by systemd/monit/manual
crashes in a couple of minutes
(goto 1)

### Log:
2019-10-07T07:46:05Z E! [agent] Error terminating process: operation not permitted
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0xf191b4]
goroutine 32 [running]:
github.com/influxdata/telegraf/plugins/inputs/exec.(*Exec).ProcessCommand(0xc0005ac0c0, 0xc0001623f1, 0x64, 0x2c59780, 0xc0005aa0a0, 0xc000552110)
#011/go/src/github.com/influxdata/telegraf/plugins/inputs/exec/exec.go:164 +0x294
created by github.com/influxdata/telegraf/plugins/inputs/exec.(*Exec).Gather
#011/go/src/github.com/influxdata/telegraf/plugins/inputs/exec/exec.go:226 +0x297
telegraf.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
telegraf.service: Unit entered failed state.
telegraf.service: Failed with result 'exit-code'.
telegraf.service: Service hold-off time over, scheduling restart.
2019-10-07T07:46:30Z I! Starting Telegraf 1.12.2
2019-10-07T07:47:05Z E! [agent] Error terminating process: operation not permitted
(same behavior again and again)

--- a/internal/config/config_test.go
+++ b/internal/config/config_test.go
@@ -158,6 +158,11 @@ func TestConfig_LoadDirectory(t *testing.T) {
 		MeasurementSuffix: "_myothercollector",
 	}
 	eConfig.Tags = make(map[string]string)
+
+	exec := c.Inputs[1].Input.(*exec.Exec)
+	require.NotNil(t, exec.Log)
+	exec.Log = nil
+
 	assert.Equal(t, ex, c.Inputs[1].Input,
 		"Merged Testdata did not produce a correct exec struct.")
 	assert.Equal(t, eConfig, c.Inputs[1].Config,

--- a/plugins/inputs/exec/exec.go
+++ b/plugins/inputs/exec/exec.go
@@ -10,13 +10,12 @@ import (
 	"sync"
 	"time"
 
-	"github.com/kballard/go-shellquote"
-
 	"github.com/influxdata/telegraf"
 	"github.com/influxdata/telegraf/internal"
 	"github.com/influxdata/telegraf/plugins/inputs"
 	"github.com/influxdata/telegraf/plugins/parsers"
 	"github.com/influxdata/telegraf/plugins/parsers/nagios"
+	"github.com/kballard/go-shellquote"
 )
 
 const sampleConfig = `
@@ -50,7 +49,7 @@ type Exec struct {
 	parser parsers.Parser
 
 	runner Runner
-	log    telegraf.Logger
+	Log    telegraf.Logger `toml:"-"`
 }
 
 func NewExec() *Exec {
@@ -161,7 +160,7 @@ func (e *Exec) ProcessCommand(command string, acc telegraf.Accumulator, wg *sync
 	if isNagios {
 		metrics, err = nagios.TryAddState(runErr, metrics)
 		if err != nil {
-			e.log.Errorf("Failed to add nagios state: %s", err)
+			e.Log.Errorf("Failed to add nagios state: %s", err)
 		}
 	}
 

--- a/plugins/inputs/exec/exec_test.go
+++ b/plugins/inputs/exec/exec_test.go
@@ -96,7 +96,7 @@ func TestExec(t *testing.T) {
 		MetricName: "exec",
 	})
 	e := &Exec{
-		log:      testutil.Logger{},
+		Log:      testutil.Logger{},
 		runner:   newRunnerMock([]byte(validJson), nil, nil),
 		Commands: []string{"testcommand arg1"},
 		parser:   parser,
@@ -126,7 +126,7 @@ func TestExecMalformed(t *testing.T) {
 		MetricName: "exec",
 	})
 	e := &Exec{
-		log:      testutil.Logger{},
+		Log:      testutil.Logger{},
 		runner:   newRunnerMock([]byte(malformedJson), nil, nil),
 		Commands: []string{"badcommand arg1"},
 		parser:   parser,
@@ -143,7 +143,7 @@ func TestCommandError(t *testing.T) {
 		MetricName: "exec",
 	})
 	e := &Exec{
-		log:      testutil.Logger{},
+		Log:      testutil.Logger{},
 		runner:   newRunnerMock(nil, nil, fmt.Errorf("exit status code 1")),
 		Commands: []string{"badcommand"},
 		parser:   parser,

